Add hash validator

kgrz opened the pull request:
Fixes #2

kgrz committed:
Add basic test cases

kgrz committed:
Add Hash Validator

- Supported options are :keys and :values that accept an Array or a Proc
  that results an Array or a method name as a symbol.

kgrz committed:
Update Readme about usage of Hash validator

kgrz committed:
Merge branch 'master' into add-hash-validator

kgrz committed:
Rename example in Readme

gnufied commented on lib/rails_legit/verify_hash_validator.rb:
```diff
@@ -0,0 +1,85 @@
+        if v.nil?
+          raise ArgumentError, "Cannot compare with nil"
+        elsif v.is_a?(Symbol)
+          if record.respond_to?(v) || record.class.private_method_defined?(v)
```
You can use `record.respond_to?(v, true)` to check for private and protected methods.

kgrz replied:
whoa! Didn't know that. Thanks. :D

kgrz committed:
Use alternate version of respond_to? method

- The `respond_to?` method accepts an optional parameter that, if set
  to `true, will make sure that private and protected methods are
  checked. No need to use the `private_method_defined?` method
